### PR TITLE
docs: rewrite git-refs readme

### LINKED ISSUE #24543: Rewrite `lib/modules/datasource/git-refs/readme.md`
### Describe the proposed change(s).

The `git-refs/readme.md` needs some work. I'll do this after:

- #24451

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Use simple words and phrases
- Separate conditions into their own paragraphs (even if that means only having one sentence per paragraph...)

## Context

Closes #24543.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/git-refs/readme.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
-This datasource can be used in combination with [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to keep dependencies up-to-date which are not specifically supported by Renovate.
+You can use this datasource plus [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to update dependencies that are not supported by Renovate.
+
+The `git-refs` datasource returns a reference from a Git repository.
```
Comment (HonkingGoose) [inline lib/modules/datasource/git-refs/readme.md:3]: > The `git-refs` datasource returns a reference from a Git repository.

Do we mean reference _from_ or reference _to_ here?

Comment (rarkins) [reply lib/modules/datasource/git-refs/readme.md:3]: From

References are a thing in git repos: https://git-scm.com/book/en/v2/Git-Internals-Git-References

File: lib/modules/datasource/git-refs/readme.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
-This datasource can be used in combination with [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to keep dependencies up-to-date which are not specifically supported by Renovate.
+You can use this datasource plus [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to update dependencies that are not supported by Renovate.
```
Comment (HonkingGoose) [inline lib/modules/datasource/git-refs/readme.md:1]: The current sentence sounds like: "I can use `git-refs` to update _any_ dependency I want". `git-refs` probably only updates _one_ or a _few_ dependencies?

In any case we should explain _which_ dependencies you can update with `git-refs`. 😄

Comment (rarkins) [inline lib/modules/datasource/git-refs/readme.md:1]:
```suggestion
You can use this datasource plus [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to update git-based dependencies that are not natively supported by Renovate.
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**